src/Transactions/Serializer.php

```diff
@@ -119,20 +81,39 @@ private function serializeData(ByteBuffer $buffer, array $options = []): void
         $buffer->writeHex($payloadHex);
     }
 
-    private function serializeCommon(ByteBuffer $buffer): void
+    /**
+     * Handle the serialization of transaction data.
+     *
+     * @param ByteBuffer $buffer
+     *
+     * @return string
+     */
+    private function serializeSignatures(ByteBuffer $buffer, array $options): void
     {
-        $buffer->writeUInt8(0xff);
-        $buffer->writeUInt8($this->transaction->data['version'] ?? 0x01);
-        $buffer->writeUInt8($this->transaction->data['network'] ?? Network::version());
-
-        $buffer->writeUint32($this->transaction->data['typeGroup'] ?? TypeGroup::CORE);
-        $buffer->writeUint16($this->transaction->data['type']);
-        $buffer->writeUint64(+$this->transaction->data['nonce']);
+        $skipSignature       = $options['skipSignature'] ?? false;
+        // $skipSecondSignature = $options['skipSecondSignature'] ?? false;
+        // $skipMultiSignature  = $options['skipMultiSignature'] ?? false;
 
-        if ($this->transaction->data['senderPublicKey']) {
-            $buffer->writeHex($this->transaction->data['senderPublicKey']);
+        if (! $skipSignature && isset($this->transaction->data['signature'])) {
+            $buffer->writeHex($this->transaction->data['signature']);
         }
 
-        $buffer->writeUint256($this->transaction->data['fee']);
+        // if (! $skipSecondSignature) {
+        //     if (isset($this->transaction->data['secondSignature'])) {
+        //         $buffer->writeHex($this->transaction->data['secondSignature']);
+        //     }
+        // }
+
+        // if (! $skipMultiSignature && isset($this->transaction->data['signatures'])) {
+        //     $buffer->writeHex(implode('', $this->transaction->data['signatures']));
+        // }
```

Comment:
can remove, unsure how multisig is going to work as it requires a contract for that